src/Resources/contao/library/Contao/InsertTags.php

```diff
@@ -562,8 +565,19 @@ protected function doReplace($strBuffer, $blnCache)
 
 				// Mobile/desktop toggle (see #6469)
 				case 'toggle_view':
-					$strUrl = ampersand(\Environment::get('request'));
+					$strRequest = \Environment::get('request');
+
+					// ESI request
+					if (preg_match('/^' . preg_quote(ltrim(\System::getContainer()->getParameter('fragment.path'), '/'), '/') . '/', $strRequest))
```

Comment:
`Environment::get('request')` now returns `_fragment?......&insertTag=....&toggle_view=desktop` which is not the link you want to show ;-) You want to add this to the original master request which is why this is passed on as query param to the ESI request.